### DISCUSSION TITLE
🐞 Can't combine Regular queries & Query params via Pydantic model

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi/fastapi#12379] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Query
from pydantic import BaseModel, NonNegativeInt, PositiveInt

app = FastAPI()


# Reusable accross multiple endpoints
class PaginationParams(BaseModel):
    offset: NonNegativeInt = 0
    limit: PositiveInt = 10


@app.get("/")
def read_root(
    pagination_query: Annotated[PaginationParams, Query()],
    q: str | None = None,  # Specific to this endpoint
):
    return {"Hello": "World"}
```

### Description

When building large apps we might have reusable query params. They can be in different forms like individual params in router like above, functional deps or in pydantic models. When combining them in fastapi router it doesn't work as expected.
![image](https://github.com/user-attachments/assets/cf35c5db-ae3b-4ca4-b26a-ca83fe776729)

Expected:
![image](https://github.com/user-attachments/assets/c3dc6bdb-c48c-4af5-9ea1-348b7b2ea8be)

### Operating System

Linux

### Operating System Details

Ubuntu 20.04.6 LTS

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

Python 3.12.5

### Additional Context

Adding docs will be great for this as well on how to combine multiple types of queries

**Comment by JavierSanchezCastro** [comment]
https://github.com/fastapi/fastapi/discussions/12236

**Reply by jd-solanki** [reply]
I wonder why I couldn't find those two 😅

**Reply by jd-solanki** [reply]
Thanks Man for both of the discussions 🙏🏻
